docs/enhancements/backup-restore-support.md

```diff
@@ -0,0 +1,152 @@
+* Spoke clusters restored on a passive hub with ACM/OADP functionality.
+* Spoke clusters moved between hubs using Kube-API.
+
+### Non-Goals
+
+* Improve import day2 cluster solution by getting full host data (e.g. using an agent on existing nodes).
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an Infrastructure Admin, I want to backup an installed spoke cluster from an active hub, including the associated BMHs (e.g. by using ACM/Kube-API/GitOps/etc).
+Then, I want to safely restore the cluster on a passive hub, including the BMHs that should keep the same state. I.e. the restored cluster should behave exactly as in the active hub with no restrictions (support day2 operations, etc).
+Note: if unbound hosts are included, they should still be automatically reprovisioned using a new discovery ISO.
+
+#### Story 2
+
+As an Infrastructure Admin, I want to backup an installed spoke cluster that doesn't have associated BMHs. Then, I want to safely restore the cluster on a passive hub.
+Note: if unbound hosts are included, they should be manually booted using a new discovery ISO. 
+
+#### Story 3
+
+As an Infrastructure Admin, I want to restore a spoke cluster in non-installed status (draft/in-progress/etc). The cluster on target hub should, ideally, keep the same status. Or, at least, fail gracefully and let the user retry the installation.
+Unbound hosts should be treated as previous stories (depends on BMHs).
+
+#### Story 4
+
+As an Infrastructure Admin, I want to safely move installed spoke clusters between hubs.
+Concerns are similar to previous stories.
+
+### Implementation Details/Notes/Constraints
+
+#### Solution for BMHs reconcile (due to a missing status)
+
+According to BMO documentation, the suggested approach for moving BareMetalHost from one cluster to another is using the `baremetalhost.metal3.io/status` [annotation](https://github.com/metal3-io/metal3-docs/blob/main/docs/user-guide/src/bmo/status_annotation.md).
+I.e. Extract the current status of the BMH, store it in an annotation, create the BMH on the new hub:
+```
+# Save the status in json format to a file
+kubectl get bmh <name-of-bmh> -o jsonpath="{.status}" > status.json
+# Save the BMH and apply the status annotation to the saved BMH.
+kubectl -n metal3 annotate bmh <name-of-bmh> \
+  baremetalhost.metal3.io/status="$(cat status.json)" \
+  --dry-run=client -o yaml > bmh.yaml
+```
+This flow ensures that the BMH on the new hub has status (BMO applies the status using the annotation). Hence, as the BMH should be in state 'Provisioned', the BMH reconcile is avoided.
+I.e. prevent destruction of restored spoke clusters.
+
+##### Suggested solution for automating the flow (make it transparent to the user):
+
+In assisted-service -> `bmh_agent_controller`:
+* If a BMH is in 'Provisioned' state (and, doesn't have `status`+`paused` annotations):
```

Comment:
In the non-converged flow the BMH moves to `Provisioned` directly after the ISO is booted IIRC

I know this is in open questions, but I figured it would be better to mention it here.

I'm not actually sure if this is a problem, but we should be sure to test both converged and non-converged if we intend this to work with both. Otherwise I think it would be okay to only support new features in converged flow, but I'm curious what @avishayt thinks about that one.